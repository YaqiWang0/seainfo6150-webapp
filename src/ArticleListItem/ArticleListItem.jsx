import React, { useState } from "react";
import ArticleTextToggleButton from "../ArticleTextToggleButton/ArticleTextToggleButton";
import ArticleImage from "../ArticleImage/ArticleImage";
import styles from "./ArticleListItem.module.css"


const ArticleListItem = (props) => {
    const[buttonText, setButtonText] = useState("Show more");

    function onClick() {
        if(buttonText === "Show more") {
            setButtonText("Show less")
        } else {
            setButtonText("Show more")
        }
    }
    return (
        <div className={styles.container}>
            <ArticleImage url = {props.article.image._url} title = {props.article.title}/>
            <div className={styles.title}>{props.article.title}</div>
            {
            buttonText === "Show less" ?
                <div>
                    <p className={styles.shortText}>{props.article.shortText}</p>
                <time className={styles.time} dateTime={props.article.timeStamp}>{props.article.displayDate}</time>
                </div>  : null
            }
            <ArticleTextToggleButton onClick = {onClick} buttonText = {buttonText}/>

        </div>
        //<DynamicArticle article = {props.article}/>
    );
};

export default ArticleListItem;