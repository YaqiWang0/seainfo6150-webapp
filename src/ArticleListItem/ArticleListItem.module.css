.container {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    border: thin solid #c7090b;
    border-radius: 15px;
    margin: 12px;
    padding-left: 10px;
    padding-right: 10px;
    padding-bottom: 10px;
    background-image: linear-gradient(to right, rgba(199, 9, 11, 0.7), white 20%);
}

.title {
    font-family: sans-serif;
    color: #c7090b;
    font-size: 1.3em;
    font-weight: bold;
    padding-top: 5px;
    text-decoration: underline;
}

.time {
    font-family: sans-serif;
    font-size: 0.6em;
    font-weight: bold;
}

.shortText {
    font-family: sans-serif;
    font-size: 0.6em;
}

