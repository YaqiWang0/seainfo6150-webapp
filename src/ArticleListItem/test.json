{
  "displayDate": "November 19th 2018, 3:10 am",
  "shortText": "I was sitting in an aisle seat on a flight earlier this year when a fellow traveler approached and politely said he thought I may be sitting in his seat.",
  "slug": "120103104851-couple-airplane-cabin",
  "timeStamp": "2018-11-19T11:10:20.000Z",
  "title": "Airline etiquette: Tips from an expert ",
  "image": {
    "_url" : "https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
  }
}