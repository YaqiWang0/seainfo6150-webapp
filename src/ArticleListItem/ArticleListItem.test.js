import React from "react";
import data from "../ArticleListItem/test";
import ArticleListItem from "./ArticleListItem";
import {fireEvent} from "@testing-library/react";

describe("ArticleListItem tests", () => {
    it("renders correctly", () => {
        const { container  } = render(<ArticleListItem article = {data}/>);
        expect(container ).toMatchSnapshot();
    });

    it("renders correctly after button is clicked", () => {
        const { container, getByText } = render(
           <ArticleListItem article = {data}/>
        );
        const buttonShowMore = getByText("Show more");
        fireEvent.click(buttonShowMore);
        expect(container).toMatchSnapshot();

        const buttonShowLess = getByText("Show less");
        fireEvent.click(buttonShowLess);
        expect(container).toMatchSnapshot();
    });
});