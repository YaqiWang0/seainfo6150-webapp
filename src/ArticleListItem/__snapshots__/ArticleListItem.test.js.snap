// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <img
        alt="Airline etiquette: Tips from an expert "
        class="image"
        src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
      />
    </div>
    <div
      class="title"
    >
      Airline etiquette: Tips from an expert 
    </div>
    <button
      class="button"
    >
      Show more
    </button>
  </div>
</div>
`;

exports[`ArticleListItem tests renders correctly after button is clicked 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <img
        alt="Airline etiquette: Tips from an expert "
        class="image"
        src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
      />
    </div>
    <div
      class="title"
    >
      Airline etiquette: Tips from an expert 
    </div>
    <div>
      <p
        class="shortText"
      >
        I was sitting in an aisle seat on a flight earlier this year when a fellow traveler approached and politely said he thought I may be sitting in his seat.
      </p>
      <time
        class="time"
        datetime="2018-11-19T11:10:20.000Z"
      >
        November 19th 2018, 3:10 am
      </time>
    </div>
    <button
      class="button"
    >
      Show less
    </button>
  </div>
</div>
`;

exports[`ArticleListItem tests renders correctly after button is clicked 2`] = `
<div>
  <div
    class="container"
  >
    <div>
      <img
        alt="Airline etiquette: Tips from an expert "
        class="image"
        src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
      />
    </div>
    <div
      class="title"
    >
      Airline etiquette: Tips from an expert 
    </div>
    <button
      class="button"
    >
      Show more
    </button>
  </div>
</div>
`;
